comment/utils.py

```diff
@@ -84,13 +84,31 @@ def is_comment_admin(user):
 
 
 def is_comment_moderator(user):
-    if settings.COMMENT_FLAGS_ALLOWED:
+    if settings.COMMENT_FLAGS_ALLOWED or settings.COMMENT_ALLOW_BLOCKING_USERS:
         return user.groups.filter(name="comment_moderator").exists() or user.has_perm(
             "comment.delete_flagged_comment"
         )
     return False
 
 
+def can_moderate_flagging(user):
+    return bool(
+        settings.COMMENT_FLAGS_ALLOWED and
+        (is_comment_admin(user) or is_comment_moderator(user))
+    )
+
+
+def can_moderator_block_users(moderator):
+    return settings.COMMENT_ALLOW_MODERATOR_PERFORM_BLOCKING and is_comment_moderator(moderator)
+
+
+def can_block_user(user):
+    return bool(
+        settings.COMMENT_ALLOW_BLOCKING_USERS and
```

Comment:
i think you can drop the `settings` check here because it is already performed in the `is_comment_moderator` function.

Reply:
but it doesn't for `is_comment_admin`

Reply:
You are right!

Reply:
I would say it safer to keep it. This check might be removed from `is_comment_moderator` in the future for some reason.

Reply:
you might not need it(YAGNI)